compiler/daml-lf-tools/src/DA/Daml/LF/TypeChecker/Check.hs

```diff
@@ -378,101 +380,145 @@ typeOfTmLam (var, typ) body = do
 typeOfTyLam :: MonadGamma m => (TypeVarName, Kind) -> Expr -> m Type
 typeOfTyLam (tvar, kind) expr = TForall (tvar, kind) <$> introTypeVar tvar kind (typeOf expr)
 
-typeOfAlts :: MonadGamma m => (CaseAlternative -> m Type) -> [CaseAlternative] -> m Type
+-- | Type to track which constructor ranks have be covered in a pattern matching.
+data MatchedRanks = AllRanks | SomeRanks !(Set.Set Int)
```

Comment:
This is the monoid for `Set Int` with an additional _absorbing_ element (aka a _zero_). I know that `Maybe (Set Int)` would give us the monoid for `Set Int` with an additional _neutral_ element. Is there something in `base` for an absorbing element as well?

Reply:
Definitely not in base. There are some libraries, e.g., https://hackage.haskell.org/package/monoid-absorbing-0.1.0.4/docs/Data-Monoid-Zero.html but I don’t think it’s worth the dependency.